darts/dataprocessing/transformers/mappers.py

```diff
@@ -0,0 +1,84 @@
+"""
+Mappers
+-------
+"""
+import numpy as np
+import pandas as pd
+
+from typing import Callable, Optional, Union, Sequence
+from inspect import signature
+
+from darts.timeseries import TimeSeries
+from darts.dataprocessing import Validator
+from darts.dataprocessing.transformers import BaseDataTransformer, InvertibleDataTransformer
+from darts.logging import get_logger, raise_log
+
+logger = get_logger(__name__)
+
+
+class Mapper(BaseDataTransformer[TimeSeries]):
+    def __init__(self,
+                 fn: Union[Callable[[np.number], np.number], Callable[[pd.Timestamp, np.number], np.number]],
+                 name: str = "Mapper",
+                 validators: Optional[Sequence[Validator]] = None):
+        """
+        Data transformer to apply a function to a time series (similar to calling `series.map()`)
+
+        Parameters
+        ----------
+        fn
+            Either a function which takes a value and returns a value ie. f(x) = y
+            Or a function which takes a value and its timestamp and returns a value ie. f(timestamp, x) = y
+        name
+            A specific name for the transformer
+        validators
+            Sequence of validators that will be called before transform()
+        """
+        if not isinstance(fn, Callable):
+            raise_log(TypeError("fn should be callable"), logger)
+        if len(signature(fn).parameters) not in [1, 2]:
+            raise_log(TypeError("fn must either take one or two parameters"))
+
+        super().__init__(name=name, validators=validators)
+        self._fn = fn
+
+    def transform(self, data: TimeSeries, *args, **kwargs) -> TimeSeries:
+        super().transform(data)
+        return data.map(self._fn)
+
+
+class InvertibleMapper(InvertibleDataTransformer[TimeSeries]):
+    def __init__(self,
+                 fn: Union[Callable[[np.number], np.number], Callable[[pd.Timestamp, np.number], np.number]],
+                 inverse_fn: Union[Callable[[np.number], np.number], Callable[[pd.Timestamp, np.number], np.number]],
+                 name: str = "InvertibleMapper",
+                 validators: Optional[Sequence[Validator]] = None):
+        """
+        Data transformer to apply a function and its inverse to a time series (similar to calling `series.map()`)
+
+        Parameters
+        ----------
+        fn
+            Either a function which takes a value and returns a value ie. f(x) = y
+            Or a function which takes a value and its timestamp and returns a value ie. f(timestamp, x) = y
```

Comment:
You should also document `inverse_fn`.

Reply:
> @LeoTafti It would be nice to avoid having to use lambdas wherever we could go around it. I believe it should work for builtin functions (and e.g. `math.sin`):
> 
> ```
> In [1]: len(signature(math.sin).parameters)
> Out[1]: 1
> ```
Something weird is happening there. I also get this on my machine, but [this run on GitHub](https://github.com/unit8co/darts/runs/1246785396) failed due to `math.sin` and I had to wrap it inside a lambda…

Reply:
that's weird indeed. I tried on Python 3.8, maybe 3.6 causes issues?

Reply:
Good call, doesn't work on Python 3.6 but it does work with 3.7 and 3.8…

Reply:
+1 for going the way you recommend